scrapers_next/il/people.py

```diff
@@ -0,0 +1,206 @@
+
+            line = line.text_content().strip()
+            if (
+                line == "Springfield Office:"
+                or line == "District Office:"
+                or line == ""
+                or line == "Additional District Addresses"
+                or line.startswith("Senator")
+                or line.startswith("Representative")
+                or line.startswith("Years served:")
+            ):
+                line_number += 1
+                continue
+
+            if not capitol_addr:
+                capitol_addr = line
+                line_number += 1
+            elif line_number in [2, 3] and not line.startswith("("):
+                capitol_addr += " "
+                capitol_addr += line
+                line_number += 1
+            elif not cap_phone and line.startswith("("):
+                cap_phone = line
+                line_number += 1
+            elif (line_number in [4, 5]) and re.search(r"FAX", line):
+                capitol_fax = re.search(r"(.+)\sFAX", line).groups()[0]
+                line_number += 1
+            elif not district_addr:
+                district_addr = line
+                line_number += 1
+                district_addr_lines += 1
+            elif district_addr_lines == 1:
+                district_addr += " "
+                district_addr += line
+                line_number += 1
+                district_addr_lines += 1
+            elif (
+                district_addr_lines == 2
+                and not line.strip().startswith("(")
+                and not re.search(r"Email:", line)
+            ):
+                district_addr += " "
+                district_addr += line
+                line_number += 1
+                district_addr_lines += 1
+            elif not dis_phone:
+                dis_phone = line
+                line_number += 1
+            elif re.search(r"FAX", line):
+                dis_fax = re.search(r"(.+)\sFAX", line).groups()[0]
+                line_number += 1
+            elif re.search(r"Email:\s", line.strip()):
+                email = re.search(r"Email:\s(.+)", line).groups()[0].strip()
+                line_number += 1
+
+        return (
+            capitol_addr,
+            cap_phone,
+            capitol_fax,
+            district_addr,
+            dis_phone,
+            dis_fax,
+            email,
+            # additional_office,
+            # additional_phone
+        )
+
+
+class LegList(HtmlListPage):
+    selector = CSS("form table tr")
+
+    def process_item(self, item):
+        # skip header rows
+        if (
+            len(CSS("td").match(item)) == 1
+            or CSS("td").match(item)[0].get("class") == "header"
+        ):
+            self.skip()
+
+        first_link = CSS("td a").match(item)[0]
+        name = first_link.text_content()
+        detail_link = first_link.get("href")
+
+        # these are "Former Senate Members"
+        former_members = [
```

Comment:
is there a way to detect these from the page?  the hard coded list will get out of date quickly